src/pages/ArticleList.jsx

```diff
@@ -1,21 +1,31 @@
 import { useEffect, useState } from "react";
 import { getAllArticles } from "../api";
 import ArticleCard from "../components/ArticleCard";
+import { useParams } from "react-router-dom";
 
 const ArticleList = () => {
+  const { slug } = useParams();
   const [isLoading, setIsLoading] = useState(true);
   const [articles, setArticles] = useState([]);
 
   useEffect(() => {
-    getAllArticles().then(({ articles }) => {
+    const query = new URLSearchParams;
+
+    if (slug) {
+      query.set('topic', slug);
+    }
+
+    getAllArticles(query).then(({ articles }) => {
       setArticles(articles);
       setIsLoading(false);
     });
-  }, []);
+  }, [slug]);
```

Comment:
We could do with some error handling when fetching the articles. At the moment if I lose connection it hangs in a loading state.

Reply:
Thanks Stephen, really appreciate your feedback!  😀